[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer-k8s/0 [executing] active: 
  pgbouncer-k8s/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer-k8s/0 [idle] active: 
  pgbouncer-k8s/1 [idle] active: 
[32mINFO    [0m httpx:_client.py:1740 HTTP Request: DELETE https://10.1.0.20:16443/api/v1/namespaces/testing/pods/pgbouncer-k8s-0 "HTTP/1.1 200 OK"
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql-test-app/0 [idle] active: 
  postgresql-k8s/0 [idle] active: Primary
  postgresql-k8s/1 [idle] active: 
  pgbouncer-k8s/0 [executing] maintenance: stopping charm software
  pgbouncer-k8s/1 [idle] active: 
  secondary-application/0 [idle] active: received database credentials of the second database
  finos-waltz-k8s/0 [idle] active: 
  pgbouncer-k8s-secondary/0 [idle] active: 
  data-integrator/0 [idle] active: 
  redis-broker/0 [idle] active: 
  redis-cache/0 [idle] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer-k8s/0 [idle] active: 
  pgbouncer-k8s/1 [idle] active:
[32mINFO    [0m pytest_operator.plugin:plugin.py:937 Model status:

Model    Controller          Cloud/Region        Version  SLA          Timestamp
testing  concierge-microk8s  microk8s/localhost  3.6.5    unsupported  02:01:43Z

App                      Version               Status  Scale  Charm                Channel        Rev  Address         Exposed  Message
data-integrator                                active      1  data-integrator      latest/edge    147  10.152.183.101  no       
finos-waltz-k8s                                active      1  finos-waltz-k8s      latest/edge     33  10.152.183.100  no       
pgbouncer-k8s            1.21.0                active      2  pgbouncer-k8s                         0  10.152.183.161  no       
pgbouncer-k8s-secondary  1.21.0                active      1  pgbouncer-k8s                         1  10.152.183.118  no       
postgresql-k8s           14.17                 active      2  postgresql-k8s       14/edge        581  10.152.183.219  no       
postgresql-test-app                            active      1  postgresql-test-app  latest/edge    379  10.152.183.138  no       
redis-broker             ubuntu/redis@691f315  active      1  redis-k8s            latest/stable    7  10.64.140.43    no       
redis-cache              ubuntu/redis@691f315  active      1  redis-k8s            latest/stable    7  10.64.140.44    no       
secondary-application                          active      1  postgresql-test-app  latest/edge    379  10.152.183.165  no       received database credentials of the second database

Unit                        Workload  Agent  Address      Ports     Message
data-integrator/0*          active    idle   10.1.152.21            
finos-waltz-k8s/0*          active    idle   10.1.152.18            
pgbouncer-k8s-secondary/0*  active    idle   10.1.152.19            
pgbouncer-k8s/0*            active    idle   10.1.152.29            
pgbouncer-k8s/1             active    idle   10.1.152.20            
postgresql-k8s/0*           active    idle   10.1.152.14            Primary
postgresql-k8s/1            active    idle   10.1.152.15            
postgresql-test-app/0*      active    idle   10.1.152.9             
redis-broker/0*             active    idle   10.1.152.26  6379/TCP  
redis-cache/0*              active    idle   10.1.152.28  6379/TCP  
secondary-application/0*    active    idle   10.1.152.17            received database credentials of the second database

[32mINFO    [0m pytest_operator.plugin:plugin.py:943 Juju error logs:

unit-postgresql-test-app-0: 01:43:58 ERROR unit.postgresql-test-app/0.juju-log Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/src/charm.py", line 560, in <module>
    main(ApplicationCharm)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/lib/python3.10/site-packages/ops/__init__.py", line 347, in __call__
    return _main.main(charm_class=charm_class, use_juju_for_storage=use_juju_for_storage)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/lib/python3.10/site-packages/ops/_main.py", line 565, in main
    manager.run()
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/lib/python3.10/site-packages/ops/_main.py", line 550, in run
    self._emit()
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/lib/python3.10/site-packages/ops/_main.py", line 502, in _emit
    self._emit_charm_event(self.dispatcher.event_name)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/lib/python3.10/site-packages/ops/_main.py", line 537, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/lib/python3.10/site-packages/ops/framework.py", line 348, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/lib/python3.10/site-packages/ops/framework.py", line 907, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/venv/lib/python3.10/site-packages/ops/framework.py", line 997, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-postgresql-test-app-0/charm/src/charm.py", line 489, in _on_run_sql_action
    cursor.execute(query)
psycopg2.errors.ReadOnlySqlTransaction: cannot execute DROP TABLE in a read-only transaction

unit-pgbouncer-k8s-0: 01:45:49 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-k8s-0: 01:45:53 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-k8s-1: 01:45:53 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-test-app-0: 01:46:39 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-pgbouncer-k8s-0: 01:47:58 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-k8s-1: 01:51:31 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-k8s-0: 01:51:31 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-pgbouncer-k8s-0: 01:53:05 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-pgbouncer-k8s-secondary-0: 01:54:29 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-finos-waltz-k8s-0: 01:54:30 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-pgbouncer-k8s-0: 01:55:28 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-pgbouncer-k8s-secondary-0: 01:58:33 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-finos-waltz-k8s-0: 01:58:34 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-data-integrator-0: 01:59:02 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-pgbouncer-k8s-1: 01:59:30 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-pgbouncer-k8s-0: 01:59:30 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
application-redis-cache: 01:59:40 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-pgbouncer-k8s-0: 02:00:03 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-pgbouncer-k8s-0: 02:00:58 ERROR unit.pgbouncer-k8s/0.juju-log Cluster upgrade failed, ensure pre-upgrade checks are ran first.
unit-pgbouncer-k8s-1: 02:01:03 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent

[32mINFO    [0m pytest_operator.plugin:plugin.py:1025 Forgetting model main...